src/ophys_etl/transforms/sine_dewarp.py

```diff
@@ -0,0 +1,459 @@
+import numpy as np
+import os
+import h5py
+import math
+import time
+import allensdk.core.json_utilities as ju
+import argparse
+import logging
+import multiprocessing
+import functools
+
+
+# Default parameters
+mean_modevalue = 0.0
+stdv_modevalue = 0.0
+bgfactorL = 0.0
+bgfactorR = 0.0
+L = 0
+R = 0
+
+minvalue = 0.0
+maxlevel = 65535  # 16bit
+
+# Default is 2P.4 parameters
+aL = 160.0
+aR = 160.0
+bL = 95.0
+bR = 110.0
+
+valid_nr_methods = [0, 1, 2, 3]
+
+
+def mode(data):
```

Comment:
This probably is a good candidate for a refactor + unit test. It looks like it's just getting the mode (where there can be more than 1). 

Try [collections.Counter](https://docs.python.org/3.7/library/collections.html#collections.Counter) or [scipy.stats.mode](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.mode.html)

Reply:
Looking at where this is used, I'm not sure why it's returning a list in the first place... It just takes the zero-index element of `modelist`.

Reply:
Oh shoot, I forgot to push my most recent changes before asking everyone to take a look at it. This was the one method I could easily write tests for haha.

I assume that it doesn't use any of those other methods because those packages aren't imported anywhere. Do we want to add a dependency like that just for their mode method?

You're right though that it doesn't need to return a list. I'm going to guess that someone just googled an algorithm for this and copied it and kept the full functionality just in case they ever needed it.

This won't currently be imported anywhere else since it is just now being moved into this repo from another one.